packml_ros/include/packml_ros/packml_ros.h

```diff
@@ -40,12 +42,17 @@ class PackmlRos
   std::shared_ptr<packml_sm::AbstractStateMachine> sm_;
   ros::Publisher status_pub_;
   ros::ServiceServer trans_server_;
+  ros::ServiceServer reset_stats_server_;
+  ros::ServiceServer get_stats_server_;
   packml_msgs::Status status_msg_;
+  packml_sm::PackmlStatsProvider stats_provider_;
```

Comment:
What about the stats provider requires it to be a ROS specific implementation?  Shouldn't it be generic enough to capture the stats at a generic level?  Shouldn't the ROS implementation just perform the conversion to ROS messages/services?

Reply:
Nothing, there is nothing ROS specific in it, other than a helper file for ROS implementations.